### PR TITLE
feat: added config for new typescript plugin

### PR DESCRIPTION
**Pull request by ashwinibm**
### Summary 
Created new plugin for TS related linting
Used by [PR](https://github.com/Instawork/hyperview/pull/509/)
Updated custom rules to match TS files also

Also had to update Node version for CircleCI since the newly added `@typescript-eslint/eslint-plugin` was incompatible with the existing version. Ref: [error](https://app.circleci.com/pipelines/github/Instawork/eslint-plugin-instawork/79/workflows/7bd959d5-225f-4fdd-bc9a-0731847737f6/jobs/169?invite=true#step-102-6)

### CODE REVIEW

**Review by flochtililoch — COMMENTED** [review]
How do you envision us using the new recommended-typescript file? I assume both recommended.js and recommended-typescript.js would need to be imported in projects, but what I'm not clear on, is whether both would apply to TS files. Since we have a lot of rules configured in recommended.js, I'm curious how we're planning on preserving these for TS code.

**Comment by ashwinibm** [comment]
> How do you envision us using the new recommended-typescript file? I assume both recommended.js and recommended-typescript.js would need to be imported in projects, but what I'm not clear on, is whether both would apply to TS files. Since we have a lot of rules configured in recommended.js, I'm curious how we're planning on preserving these for TS code.

By using overrides and file patterns, we're able to inherit the common JS config and extend with the new TS config.
See the usage [here](https://github.com/Instawork/hyperview/pull/509/files#diff-e7e195d3d8fa5b82fa51ed952f3a179d429cbc46c0432ce1f7b357c54c162822R54-R57)

Item 3 in the demo screenshots in this [PR](https://github.com/Instawork/hyperview/pull/509) shows how Instawork custom rules from the `recommended` JS plugin still get applied to `index.ts`

File: src/config/common-config.js [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+  'react/jsx-props-no-spreading': 'warn',
+  'react/jsx-sort-props': 'warn',
+  'react/no-deprecated': 'warn',
+  'react/prop-types': 'warn',
+  'react/state-in-constructor': 'warn',
+  'react/static-property-placement': 'warn',
+  'sort-keys': 'warn',
+};
+
+module.exports = {
```
Comment (ashwinibm) [inline src/config/common-config.js:124]: As discussed @flochtililoch, I've pulled out the common rule overrides.

**Review by flochtililoch — APPROVED** [review]
LGTM, thanks for the updates!

**Pull request merged**